pkg/client/client.go

```diff
@@ -36,11 +39,17 @@ func NewClient[T Client](opts ...func(T)) T {
 }
 
 func WithCertPool[T Client](certPool *x509.CertPool) func(T) {
+	// make sure we have a valid cert pool
 	if certPool == nil {
 		return func(client T) {}
 	}
 	return func(client T) {
-		client.GetClient().Transport = &http.Transport{
+		// make sure that we can access the internal client
+		if client.GetInternalClient() == nil {
+			log.Warn().Msg("internal client is invalid")
```

Comment:
I see this message in the logs and am not sure what to do about it (e.g. if it's an error me as the user needs to fix or if it's normal):

```
# ./magellan -v collect --cacert /opt/shared/ochami.pem --host https://cable-guys.si.usrc --username <username> --password <password> --verbose
{"level":"warn","time":"2024-08-27T21:17:38Z","message":"internal client is invalid"}
[...]
```

it goes on to succeed, so I assume it's normal.

Could you make this a more informative message?

Reply:
It's supposed to indicate that there's something wrong internally like a bug with the implemented client and nothing for the end user to fix. I thought setting the client in `Init()` should have bypassed the check though. I'll update the message to be more informative.